drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -374,11 +436,51 @@ local function set_setpoint(setpoint)
   end
 end
 
-local function setpoint_limit_handler(limit_field)
+local heating_setpoint_limit_handler_factory = function(minOrMax)
   return function(driver, device, ib, response)
+    if ib.data.value == nil then
+      return
+    end
     local val = ib.data.value / 100.0
-    log.info("Setting " .. limit_field .. " to " .. string.format("%s", val))
-    device:set_field(limit_field, val, { persist = true })
+    if val >= 40 then -- assume this is a fahrenheit value
+      val = utils.f_to_c(val)
+    end
+    device:set_field(minOrMax, val)
+    local min = device:get_field(setpoint_limit_device_field.MIN_HEAT)
+    local max = device:get_field(setpoint_limit_device_field.MAX_HEAT)
+    if min ~= nil and max ~= nil then
+      if min < max then
+        device:emit_event_for_endpoint(ib.endpoint_id, capabilities.thermostatHeatingSetpoint.heatingSetpointRange({ value = { minimum = min, maximum = max }, unit = "C" }))
```

Comment:
The temperature conversions for tempRange (and the setpoint values as well) are supported in FW 53+. Therefore, we won't want to send these requests for cases where the hub is running FW 53 or greater. We could gate this on lua libs api version 10 since that is the api version for lua libs packaged with FW 53.

Caveat: Lua libs version until now has essentially been tied to FW version since we use the lua libs packaged in FW. However, this parity may be broken with async lua libs releases, so this is something we will need to consider. The change to support these temp conversion is not technically in the lua libs, it is in the FW.

Reply:
As discussed, added a check for RPC version in f56fc0de8347451163d76740b9781153ce65328c.